tests/Configuration/ComposerJsonParserTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Tests\Configuration;
+
+use Nette\Utils\Json;
+use Rector\Core\Configuration\MinimalVersionChecker\ComposerJsonParser;
+use Symplify\PackageBuilder\Tests\AbstractKernelTestCase;
+
+final class ComposerJsonParserTest extends AbstractKernelTestCase
+{
+    public function testExact(): void
+    {
+        $version = '7.2.0';
+        $actualPhpVersion = $this->getComposerJsonPhpVersion($version);
+
+        $this->assertSame('7.2.0', $actualPhpVersion);
+    }
+
+    public function testTilde(): void
+    {
+        $version = '~7.2.0';
+        $actualPhpVersion = $this->getComposerJsonPhpVersion($version);
+
+        $this->assertSame('7.2.0', $actualPhpVersion);
+    }
+
+    public function testStarEnd(): void
+    {
+        $version = '7.2.*';
+        $actualPhpVersion = $this->getComposerJsonPhpVersion($version);
+
+        $this->assertSame('7.2', $actualPhpVersion);
+    }
+
+    public function testStarMiddleEnd(): void
+    {
+        $version = '7.*.*';
+        $actualPhpVersion = $this->getComposerJsonPhpVersion($version);
+
+        $this->assertSame('7', $actualPhpVersion);
+    }
+
+    public function testCaret(): void
+    {
+        $version = '^7.2.0';
+        $actualPhpVersion = $this->getComposerJsonPhpVersion($version);
+
+        $this->assertSame('7.2.0', $actualPhpVersion);
+    }
+
+    public function testSimpleRange(): void
+    {
+        $version = '>=7.2.0';
+        $phpVersion = $this->getComposerJsonPhpVersion($version);
+
+        $this->assertSame('7.2.0', $phpVersion);
+    }
```

Comment:
Could you merge this into 1 test method with data provider?

Reply:
Ready